deprecate tuples as collection Recording pointers

jacobagilbert opened the pull request:
This is done in favor of JSON objects, avoiding nested vectors (prohibited by flatbuffers) and also permitting expansion within this object.

bhilburn commented on sigmf-spec.md:
```diff
@@ -663,22 +663,42 @@ this field be capable of displaying up to 20 characters.
 1. The Collection Object MUST be the only top-level object in the file.
-1. The Collection Object MUST only contain SigMF key/value pairs.
-1. Keys in the Collection Object MUST use SigMF Recording Tuples to reference 
+1. The Collection Object MUST only contain SigMF key/value pairs provided by the
```
I would remove this rule from this list, and rather add it to the rules about Compliance later in the spec.

jacobagilbert replied:
Don't disagree....but this suggestion feels unrelated to this PR.

bhilburn commented on sigmf-spec.md:
````diff
@@ -663,22 +663,42 @@ this field be capable of displaying up to 20 characters.
+```
+
+Tuples will be removed in SigMF version 2.0, so JSON Objects are RECOMMENDED.
+Additional optional user fields MAY be added to `SigMF Recording objects` if
````
`objects` should be capitalized

bhilburn commented on sigmf-spec.md:
```diff
@@ -663,22 +663,42 @@ this field be capable of displaying up to 20 characters.
+   Metadata file `[base-name].sigmf-meta`.
+1. If a value contains multiple SigMF Recordings, they MUST appear in a JSON array.
+
+`SigMF Recording Objects` reference the base-name of the SigMF Recording and the
```
This should be a sub-section with a table describing the required fields in the Object, like we do elsewhere in the spec. Reading it, I realize we should have previously done this for the tuple version, too, but neglected it at the time.

bhilburn reviewed (changes requested):
@jacobagilbert - I like this proposed solution. Good idea / design. My comments are all copy related, not design related.

bhilburn commented:
Bumping, looks like no changes since last review

jacobagilbert commented:
@bhilburn thanks for the ping. I think I have addressed most of the feedback. I also did a quick scan through this for `object` --> `Object` language and upgraded a number of uses to the proper `Object` form.

bhilburn reviewed (approved):
Minor language comment, otherwise this PR is great and is the best way forward for this issue IMO.

jacobagilbert commented on sigmf-spec.md:
```diff
@@ -689,35 +690,68 @@ The following names are specified in the `core` namespace for use in the `collec
+#### SigMF Recording Objects
+
+`SigMF Recording Objects` reference the base-name of the SigMF Recording and the
+SHA512 hash of the Metadata file, and can be specified as a JSON Object:
```
Yep, this is correct will fix and merge.

Thanks!

jacobagilbert commented:
Issues addressed, multiple approvals. Merging.
